src/Workspaces/Core/Desktop/Workspace/SQLite/Interop/Util.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.SQLite
+{
+    internal static class Util
+    {
+        internal static byte[] ToUtf8(string sourceText)
+        {
+            if (sourceText == null)
+            {
+                return null;
+            }
+
+            byte[] byteArray;
+            int nlen = Encoding.UTF8.GetByteCount(sourceText) + 1;
+
+            byteArray = new byte[nlen];
+            nlen = Encoding.UTF8.GetBytes(sourceText, 0, sourceText.Length, byteArray, 0);
+            byteArray[nlen] = 0;
```

Comment:
this should be documented why you're doing this (i presume sqlite requires 0 terminated utf8?).

Reply:
Yes.

Reply:
This was shamelessly copy pasted from sqlitepcl.raw :p

Reply:
Just to be clear.  By 'documented', i meant: 'comment in the code' :)

Reply:
Yes, I did add some comments.